cmd/kubectl-gadget/snapshot/process.go

```diff
@@ -128,32 +166,32 @@ func processTransformEvent(e types.Event) string {
 				e.Command, e.Pid))
 		}
 	case utils.OutputModeCustomColumns:
-		for _, col := range params.OutputConf.CustomColumns {
+		for _, col := range outputConf.CustomColumns {
 			switch col {
 			case "node":
-				sb.WriteString(fmt.Sprintf("%s\t", e.Node))
+				sb.WriteString(fmt.Sprintf("%s", e.Node))
 			case "namespace":
-				sb.WriteString(fmt.Sprintf("%s\t", e.Namespace))
+				sb.WriteString(fmt.Sprintf("%s", e.Namespace))
 			case "pod":
-				sb.WriteString(fmt.Sprintf("%s\t", e.Pod))
+				sb.WriteString(fmt.Sprintf("%s", e.Pod))
 			case "container":
-				sb.WriteString(fmt.Sprintf("%s\t", e.Container))
+				sb.WriteString(fmt.Sprintf("%s", e.Container))
 			case "comm":
-				sb.WriteString(fmt.Sprintf("%s\t", e.Command))
+				sb.WriteString(fmt.Sprintf("%s", e.Command))
 			case "tgid":
-				sb.WriteString(fmt.Sprintf("%d\t", e.Tgid))
+				sb.WriteString(fmt.Sprintf("%d", e.Tgid))
 			case "pid":
-				sb.WriteString(fmt.Sprintf("%d\t", e.Pid))
+				sb.WriteString(fmt.Sprintf("%d", e.Pid))
 			}
-			sb.WriteRune(' ')
+			sb.WriteRune('\t')
 		}
 	}
 
 	return sb.String()
 }
 
-func printProcesses(allProcesses []types.Event) error {
-	if !processCollectorParamThreads {
+func sortProcessEvents(allProcesses []types.Event, processFlags *ProcessFlags) {
```

Comment:
Oops. You are right! I didn't realise deep enough that functionality. I followed your suggestion about returning the new slice, but I was wondering if it would be better to use pointers 🤔

Reply:
> I followed your suggestion about returning the new slice, but I was wondering if it would be better to use pointers thinking

Yes sorry, I suggested the first thing which came to my mind to solve it but indeed using pointers would be better as it would avoid allocating a new slice.

Reply:
> it would avoid allocating a new slice
>
We still need to do it because it is easier to create a new slice where we only add the main thread per process than to remove the secondary threads from the original slice.

My only concern is that a pointer to a slice (`*[]types.Event`) doesn't look very go-style 😅 .